mlir/lib/Dialect/Linalg/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -932,13 +932,18 @@ struct PackOpTiling
           continue;
         }
 
-        // If the dimension needs padding, it is not supported because there are
-        // iterations that only write padding values to the whole tile. The
-        // consumer fusion is driven by the source, so it is not possible to map
-        // an empty slice to the tile.
-        bool needExtraPadding =
-            ShapedType::isDynamic(destDimSize) || !cstInnerSize ||
-            destDimSize * cstInnerSize.value() != srcDimSize;
+        // If the dimension needs extra padding, it is not supported because
+        // there are iterations that only write padding values to the whole
+        // tile. The consumer fusion is driven by the source, so it is not
+        // possible to map an empty slice to the tile. Extra padding is not a
+        // regular form, and the implementation is being conversative.
+        bool needExtraPadding = true;
+        if (!ShapedType::isDynamic(srcDimSize) &&
+            !ShapedType::isDynamic(destDimSize) && cstInnerSize) {
+          needExtraPadding =
+              destDimSize >
+              (srcDimSize + cstInnerSize.value() - 1) / cstInnerSize.value();
+        }
```

Comment:
This is going to be hard if we allow extra padding. E.g., it could result in dynamic shape easily and it is known that the extra padding is not required after some tiling. I'm considering to restrict the semantics a little. I'm working on a local patch, and can send an RFC to discourse.

cc @MaheshRavishankar @Max191 @egebeysel @adam-smnk @banach-space